### PR TITLE
Support astral plane characters on Windows

### PR DESCRIPTION
**Pull request by chris-morgan**

### CODE REVIEW

**Comment by cmyr** [comment]
cool this looks good, will test when I'm next in front of my proper machine. Thanks!

File: src/widget/textbox.rs [hunk 1]
```diff
@@ -431,12 +431,13 @@ impl Widget<String> for TextBoxRaw {
                         self.reset_cursor_blink(ctx);
                     }
                     // Actual typing
-                    k_e if k_e.key_code.is_printable() => {
```
Comment (raphlinus) [inline src/widget/textbox.rs]: The problem with removing this is that control characters get inserted into the string. I agree with you that we shouldn't be looking at the keycode (that's not really a source of truth), but should have an API indicating whether the keyboard event has printable text.

Comment (chris-morgan) [reply src/widget/textbox.rs]: In what situations do control characters come through this path? I haven’t tried much, but nothing immediately sprang to mind.

Comment (raphlinus) [reply src/widget/textbox.rs]: It happens when you press, say, control-A, but the command isn't interpreted.

**Comment by cmyr** [comment]
@chris-morgan do you have any interest in revisiting this?

**Comment by chris-morgan** [comment]
Not personally at present. In a month or six maybe if no one has by then, but not now. This was just a quick proof of concept, someone else can feel free to take it over and do it however they like.

**Comment by cmyr** [comment]
okay, thanks. Going to close for now, we can revisit down the road.

**Pull request closed without merging**